src/main/java/org/jabref/logic/pdf/search/indexing/IndexingTaskManager.java

```diff
@@ -39,27 +40,30 @@ protected Void call() throws Exception {
     private void enqueueTask(BackgroundTask<Void> task) {
         task.onFinished(() -> {
             this.progressProperty().unbind();
+            this.messageProperty().unbind();
             this.updateProgress(1, 1);
             taskQueue.poll(); // This is the task that just finished
             if (!taskQueue.isEmpty()) {
                 BackgroundTask<Void> nextTask = taskQueue.poll();
                 nextTask.executeWith(taskExecutor);
                 this.progressProperty().bind(nextTask.progressProperty());
+                this.messageProperty().bind(nextTask.messageProperty());
```

Comment:
It looks like you are reimplementing your own task queue system here. Why not reuse the one that is already in place? 
https://github.com/JabRef/jabref/blob/a7bd8b750395fa641e491368c59f8d438b589aaa/src/main/java/org/jabref/gui/util/DefaultTaskExecutor.java#L33
I think you can simply replace `enqueueTask(task)` by `task.executeWith(taskExecutor)`. The default task executor should already take care that not too many tasks run in parallel. If you are afraid to block other tasks that are submitted to the task executor, I would propose to add a `executeInBackgroundWith` method to the task executor, that uses it's own executor as in 
```
private final ExecutorService backgroundExecutor = Executors.newFixedThreadPool(2); 
```
The fixed thread pool executor already implements the queing system, so you don't have to do that.

Reply:
> I think you can simply replace enqueueTask(task) by task.executeWith(taskExecutor)

I guess this connects to the discussion we are having in #7961 about having one or more tasks per database. I think the placeholder-task strategy would no longer work with this approach because the scheduling is done by the ExecutorService so I cannot bind the progress and message properties when starting a task. Of course, we could do that when actually starting/finishing an operation.

> The default task executor should already take care that not too many tasks run in parallel.

The index should only be modified by one thread at a time, so I would choose a pool size of 1.

> If you are afraid to block other tasks that are submitted to the task executor, I would propose to add a executeInBackgroundWith method to the task executor, that uses it's own executor

Would having a local executor in IndexingTaskManager be acceptable?

Reply:
> The index should only be modified by one thread at a time, so I would choose a pool size of 1.

In this case, why not have one big background task that iterates over all entries and their files and index them synchronously (in the task)? If you only want to index one file at the time, then why do you need this complicated structure with the queue? 

> Would having a local executor in IndexingTaskManager be acceptable?

Then you would have to deal with notifying the indexing manager about shutdown of jabref, because otherwise the executor keeps running.

Reply:
> In this case, why not have one big background task that iterates over all entries and their files and index them synchronously (in the task)?

That's what I am doing on startup, but while JabRef is running users may add or remove files and the index needs to be updated accordingly.

> Then you would have to deal with notifying the indexing manager about shutdown of jabref, because otherwise the executor keeps running.

I'll keep that in mind, thanks!